src/Sylius/Bundle/PromotionBundle/spec/Command/GenerateCouponsCommandSpec.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+declare(strict_types=1);
+
+namespace spec\Sylius\Bundle\PromotionBundle\Command;
+
+use PhpSpec\ObjectBehavior;
+use Prophecy\Argument;
+use Sylius\Bundle\CoreBundle\Doctrine\ORM\PromotionRepository;
+use Sylius\Bundle\PromotionBundle\Command\GenerateCouponsCommand;
+use Sylius\Component\Core\Model\PromotionInterface;
+use Sylius\Component\Promotion\Generator\PromotionCouponGeneratorInstruction;
+use Sylius\Component\Promotion\Generator\PromotionCouponGeneratorInstructionInterface;
+use Sylius\Component\Promotion\Generator\PromotionCouponGeneratorInterface;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+final class GenerateCouponsCommandSpec extends ObjectBehavior
```

Comment:
PHPSpec tests for commands are not the best idea, IMO. Commands are like controllers. They should contain as little logic as possible. In this specific case, the implementation is quite nice - the command just validates the data and uses a service, so I would skip the test in this case (does not provide a considerable value) or, if we want to be 100% perfect 🖖, go with a Behat test for this feature 🏇

Reply:
I will look at how to create a behat test for it next week.

Reply:
Moved to phpunit.